pkg/api/serialization_test.go

```diff
@@ -98,9 +101,18 @@ func roundTripSame(t *testing.T, item runtime.Object, except ...string) {
 		return
 	}
 
-	version := testapi.Default.Version()
-	if !set.Has(version) {
-		fuzzInternalObject(t, version, item, seed)
+	gv := testapi.Default.GroupVersion()
+	gvk, err := api.Scheme.ObjectKind(item)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+		return
+	}
+	// if we don't recognize the type, then we'll never be able to round trip it
+	if !api.Scheme.Recognizes(gv.WithKind(gvk.Kind)) {
+		return
```

Comment:
> no error in this case?

This matches the `set` check below.  Basically, they run the same unit tests multiple times and switch group,versions each time, so the test can be run sometimes and not others.  This is the correct form.  Refactoring its location doesn't work out cleaner because it makes the structure mix and match.